aiida/cmdline/commands/cmd_import.py

```diff
@@ -25,6 +26,52 @@
 COMMENT_MODE = ['newest', 'overwrite']
 
 
+class ImportPath(click.Path):
+    """AiiDA Verdi implementation of Click's Path-type.
+    Introduce tests to distinguish whether the path is to a local file
+    or a URL string to a file that needs to be downloaded.
+    """
+
+    def __init__(self, **kwargs):
+        super(ImportPath, self).__init__(**kwargs)
+
+    def __call__(self, value, param=None, ctx=None):
+        if value is not None:
+            try:
+                res = super(ImportPath, self).convert(value, param, ctx)
+            except click.exceptions.BadParameter:
+                res = self.convert(value, param, ctx)
+            return res
+        return None
+
+    def convert(self, value, param, ctx):
+        """Do checks for possible URL path
+        Parameters :param:`readable` and :param:`exists` are always set to `True`.
+        """
+        from six.moves import urllib
+        from django.core.validators import URLValidator
+        from django.core.exceptions import ValidationError as URLValidationError
+        from socket import timeout
+
+        url = value
+        timeout_seconds = 10
```

Comment:
this should probably go as a constant in some configuration module

Reply:
Probably should. And I think the whole class should go somewhere else as well. I was kind of hoping @sphuber could help me out on this :)

Reply:
I have moved `ImportPath` to `aiida.cmdline.params.path`.
The timeout seconds are moved into this file as well.